libs/custom-paper-handler/src/driver/driver.ts

```diff
@@ -204,53 +217,238 @@ export class PaperHandlerDriver implements PaperHandlerDriverInterface {
-      // argument for `Real-time status transmission`, the printer will return 1 byte 0x12
-      // on the generic transfer-in buffer. 0x12 is an undocumented response code but
-      // presumably means "Invalid argument" or similar.
-      const IgnorableResponseCoder = oneOf(
-        RealTimeStatusTransmission,
-        InvalidArgumentErrorCode
-      );
+  private isUsbDisconnectionError(error: unknown): boolean {
+    if (!(error instanceof Error)) {
+      return false;
+    }
 
-      const decodeResult = IgnorableResponseCoder.decode(
-        bufferFromDataView(data)
+    const errorMessage = extractErrorMessage(error.message).toLowerCase();
+
+    // Specific USB disconnection error patterns from libusb
+    // Note: LIBUSB_ERROR_IO is intentionally excluded as it's too broad and
+    // can indicate other non-disconnection I/O errors
+    const disconnectionPatterns = [
+      'libusb_transfer_no_device',
+      'libusb_error_no_device',
+      'no_device',
+      'device not found',
+      'device disconnected',
+      'not connected',
+    ];
+
+    return disconnectionPatterns.some((pattern) =>
+      errorMessage.includes(pattern)
+    );
+  }
+
+  /**
+   * Attempts to reconnect to the USB device once.
+   * Uses a lock to ensure only one reconnection attempt happens at a time.
+   * Sets the isReconnecting flag to prevent infinite recursion during re-initialization.
+   */
+  private async reconnectUsbDevice(
+    originalError: unknown,
+    delayMs: number
+  ): Promise<boolean> {
+    // Use a lock to prevent multiple simultaneous reconnection attempts
+    return await this.reconnectionLock.withLock(async () => {
+      debug(
+        `Attempting to reconnect to USB device after error: ${extractErrorMessage(
+          originalError
+        )}`
       );
-      if (decodeResult.isOk()) {
+
+      try {
+        // Close the current connection if still open
+        try {
+          await this.webDevice.close();
+        } catch (closeError) {
+          // Ignore errors when closing disconnected device
+          debug(`Error closing disconnected device (expected): ${closeError}`);
+        }
+
+        // Wait before attempting to reconnect
+        debug(`Waiting ${delayMs}ms before reconnection attempt`);
+        await sleep(delayMs);
+
+        // Get a new device connection
+        const newWebDevice = await getPaperHandlerWebDevice();
+        if (!newWebDevice) {
+          debug('Failed to find device');
+          return false;
+        }
+
+        // Update the webDevice and reconnect
+        this.webDevice = newWebDevice;
+        await this.connect();
+
+        // Set flag to prevent infinite recursion during re-initialization
+        this.isReconnecting = true;
+        try {
+          // Re-initialize the device to restore settings
+          await this.initializePrinter();
+          await this.setLineSpacing(DEFAULT_LINE_SPACING);
+        } finally {
+          this.isReconnecting = false;
```

Comment:
This logic to prevent infinite recursion took me a while to understand given how it's split across multiple methods. I wonder if it would be simpler if these two re-initialization commands could just be sent as raw commands without the retry logic.